### ISSUE TITLE
Untrusted content served withouth CSP

### ISSUE THREAD

**Issue by mpkorstanje** [jenkinsci/cucumber-reports-plugin#127]
cucumber-html-reports serves files from `http://<host>/job/<project>/<build>/cucumber-html-reports/**`without CSP headers. 

However the `embeddings` folder may contain unsafe content (see [cucumber-reporting
#557](https://github.com/damianszczepanik/cucumber-reporting/pull/557))

All content from `cucumber-html-reports/embeddings` should be served with CSP headers enabled. 

Note: Enabling the CSP won't affect the iframed html as it is inlined in the srcDoc attribute.

**Comment by damianszczepanik** [comment]
I believe CSP is working fine - at least I was not able to find example where it does not. Provide sample where we have problem

**Comment by mpkorstanje** [comment]
Steps to reproduce:

Ensure the contents of a file `/var/lib/jenkins/jobs/Cucumber/builds/12/cucumber-html-reports/embeddings/` contains the following. You can do this either through running a test that creates the embedding or manipulating the file by hand.

```
<html>
   <head>
   </head>
   <body>
      <script>alert('XXS');</script>
    </body>
</html>
```
Visit that file via the browser at: `http://localhost:8080/job/Cucumber/12/cucumber-html-reports/embeddings/embedding_-1123547115.html`

![image](https://cloud.githubusercontent.com/assets/6946919/23343551/ab9ff67a-fc6d-11e6-86a7-c5cf223e2b40.png)

Observe that the CSP headers is not set and that JavaScript is allowed.

**Comment by mpkorstanje** [comment]
cucumber-html-reports version is latest build from source.

```
apt-cache policy  jenkins
jenkins:
  Installed: 2.32.2
  Candidate: 2.32.2
  Version table:
 *** 2.32.2 500
        500 http://pkg.jenkins.io/debian-stable binary/ Packages
        100 /var/lib/dpkg/status
```

**Comment by mpkorstanje** [comment]
Looking through the code I can see two solutions.

1. Don't create an embeddings folder in cucumber cucumber-reporting and embed all data. This can also be done for content that can only be downloaded.
2. Modify SafeArchiveServingAction.serveFile such that it does add the CSP headers for the `embeddings` folder. 

Both seem somewhat inelegant.

**Comment by damianszczepanik** [comment]
It was done by the purpose that the content of the plugin and its embedding is not restricted. I assume that if you generate HTML page you want to display later you should be aware of what you generate and what is the risk. And this is how it was implemented so this is the feature, not the bug.

**Comment by mpkorstanje** [comment]
It grants access to the Jenkins session cookie. If a user looks at at an malicious page that javascript will be able to do anything that user can. Preventing this is in line with the description of why the CSP policy was instated.

>  https://wiki.jenkins-ci.org/display/JENKINS/Configuring+Content+Security+Policy
> Jenkins 1.641 / Jenkins 1.625.3 introduce the Content-Security-Policy header to static files served by Jenkins (specifically, DirectoryBrowserSupport). This header is set to a very restrictive default set of permissions to protect Jenkins users from malicious HTML/JS files in workspaces, /userContent, or archived artifacts.

This may seem unimportant when there is only a single user (why do such stupid things?) However when there are multiple users it allows a malicious user without jenkins administrator rights to gain these.

A very simple scenario might go like this:

1. Malicious user M creates a report that contains malicious content.
2. Through social engineering M asks administrative user A to visit the page.
3. Either the javascript in the malicious page sends the session cookie of A to M.
4. M can now login as A and use the credentials stored Jenkins to upload a malicious artifact to a maven repository compromising downstream projects.

The feature itself -looking at html files directly- is not needed as they are displayed safely in the iframe and can be downloaded for local viewing. Currently you also have to make some effort to view the files this way. Clicking the download button will activate the dialogue, only by middle mouse clicking you can view the file directly.

**Comment by damianszczepanik** [comment]
I'm for the second solution (should be excluded in ``SafeArchiveServingRunAction``) but then the question is if there should be message that the content might be displayed improperly?
Would you prepare PR for this?

**Comment by mpkorstanje** [comment]
I'll give it a shot. I'm expecting to be able to do this in the next 3-4 weeks.

I agree on the message, but I think we should address that at the iframe in the report. That should catch more people

**Comment by damianszczepanik** [comment]
Fixed in #128
